File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    
    int plantableFront(vector<int> flowerbed) {
        int count = 0;
        int len = flowerbed.size();
        
        for (int i = 0; i < len; i++) {
            if (flowerbed[i] == 1) continue;
            if (i == 0) {
                if (flowerbed[i + 1] == 0) {
                    count++;
                    flowerbed[i] = 1;
                }
                continue;
            }
            if (i == len - 1) {
                if (flowerbed[i - 1] == 0) {
                    count++;
                    flowerbed[i] = 1;
                }
                continue;
            }
            if (flowerbed[i - 1] == 0 && flowerbed[i + 1] == 0) {
                count++;
                flowerbed[i] = 1;
            }
        }
        return count;
    }
    
    int plantableBack(vector<int> flowerbed) {
        int count = 0;
        int len = flowerbed.size();
        
        for (int i = len - 1; i >= 0; i--) {
            if (flowerbed[i] == 1) continue;
            if (i == 0) {
                if (flowerbed[i + 1] == 0) {
                    count++;
                    flowerbed[i] = 1;
                }
                continue;
            }
            if (i == len - 1) {
                if (flowerbed[i - 1] == 0) {
                    count++;
                    flowerbed[i] = 1;
                }
                continue;
            }
            if (flowerbed[i - 1] == 0 && flowerbed[i + 1] == 0) {
                count++;
                flowerbed[i] = 1;
            }
        }
        return count;
    }
    
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        
        if (flowerbed.size() == 1 && flowerbed[0] == 0) {
            if (n <= 1) return true;
            else return false;
        }
        
        int maxPlantable = max(plantableFront(flowerbed), plantableBack(flowerbed));
        if (n <= maxPlantable) {
            return true;
        }
        return false;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.kt
class Solution {
    fun canBePlaced(flowerbed: IntArray, position: Int, e: Int): Boolean {
        if (position == 0)
            if (flowerbed[0] == 0 && flowerbed[1] == 0) return true else return false
        if (position == e)
            if (flowerbed[e] == 0 && flowerbed[e - 1] == 0) return true else return false
        
        if (flowerbed[position] == 0 && flowerbed[position - 1] == 0 && flowerbed[position + 1] == 0) return true
        return false
    }
    
    fun canPlaceFlowers(flowerbed: IntArray, n: Int): Boolean {
        val e = flowerbed.size - 1
        var remaining: Int = n
        
        if (e == 0) {
            if (flowerbed[0] == 0) {
                if (n == 0 || n == 1) return true
                return false
            }
            if (flowerbed[0] == 1) {
                if (n == 0) return true
                return false
            }
        }
        
        for (i in flowerbed.indices) {
            if (canBePlaced(flowerbed, i, e)) {
                remaining--
                flowerbed[i] = 1
            }
        }
        
        if (remaining <= 0) return true
        return false
    }
}
